src/main/java/HomeScreenView.java

```diff
@@ -1,43 +1,141 @@
+import entities.Restaurant;
 import restaurant_screens.ChoicesController;
+import restaurant_screens.ChoicesSortedView;
+import restaurant_use_case.ChoicesRequestModel;
+import restaurant_use_case.ChoicesResponseModel;
+import restaurant_use_case.sortChoices;
 
 import javax.swing.*;
+import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Objects;
 
 public class HomeScreenView extends JFrame implements ActionListener {
+    /**
+     * The search query
+     */
+    JTextField query;
+    /**
+     * The input location (Postal Code)
+     */
+    JTextField location;
+    /**
+     * The selected price bucket (pricing /10)
+     */
+    JComboBox<Integer> priceBucket;
+    /**
+     * The select star rating (/5 stars)
+     */
+    JComboBox<Integer> avgStars;
+    /**
+     * The selected cuisine type
+     */
+    JComboBox<String> cuisineType;
+    /**
+     * The group of radiobuttons for the sorting selection
+     */
+    ButtonGroup sortButtons;
+    /**
+     * The choices controller
+     */
+    ChoicesController choicesController;
 
-    // Search Field(s) Temporarily placing Location as an input field
-    JTextField searchBox = new JTextField();
-    JTextField locationBox = new JTextField();
+    public HomeScreenView(ChoicesController choicesController) {
 
-    // Drop-Down Menus Options
-    Integer[] pricingOptions = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
-    Integer[] avgStarsOptions = { 1, 2, 3, 4, 5};
+        this.choicesController = choicesController;
 
-    // TODO: add more cuisine options
-    String[] cuisineOptions = { "Food"};
+        // Search Field(s) Temporarily placing Location as an input field
+        JPanel searchField = new JPanel();
 
+        query = new JTextField();
+        location = new JTextField();
 
-    // Drop-Down Menus
-    JComboBox pricingDropDown = new JComboBox(pricingOptions);
-    JComboBox avgStartsDropDown = new JComboBox(avgStarsOptions);
-    JComboBox cuisineDropDown = new JComboBox(cuisineOptions);
+        searchField.add(query);
 
-    // Sort Method Buttons
-    JRadioButton sortPriceButton = new JRadioButton();
+        // Search Button
+        JButton searchButton = new JButton("Search");
+        searchField.add(searchButton);
+        searchButton.addActionListener(this);
 
-    JRadioButton sortAvgStarsButton = new JRadioButton();
+        // Drop-Down Menus Options
+        Integer[] pricingOptions = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
```

Comment:
Currently the implemented priceBucket is on a scale of 0-5 but this can be changed

Reply:
I'll make this change. 0-5 is what I originally had but I think we flip-flopped between the range in lab last time, but yeah, it's easy to change.